execute/plugin.go

```diff
@@ -438,6 +455,11 @@ func selectReport(
 	accumulatedSize := 0
 	var finalReports []cciptypes.ExecutePluginReportSingleChain
 	for reportIdx, report := range reports {
+		// Reports at the end may not have messages yet.
+		if len(report.Messages) == 0 {
+			break
+		}
```

Comment:
Empty commit reports are expected due to the two phase report/message reading. Break out of processing rather than erroring.